lib/logstash/plugin_mixins/rabbitmq_connection.rb

```diff
@@ -48,6 +48,12 @@ def setup_rabbitmq_connection_config
         # Time in seconds to wait before retrying a connection
         config :connect_retry_interval, :validate => :number, :default => 1
 
+        # The default connection timeout; zero means wait indefinitely
+        config :connection_timeout, :validate => :number
```

Comment:
Well, my bad. Apparently the [default](http://www.rabbitmq.com/javadoc/constant-values.html#com.rabbitmq.client.ConnectionFactory.DEFAULT_CONNECTION_TIMEOUT) connection timeout is zero! I'll have it behave as you suggested then.